src/NBS.Appointments.Service/Models/AvailabilityHourResponse.cs

```diff
@@ -40,28 +40,37 @@ public static AvailabilityHourResponse FromQflowResponse(SiteSlotsResponse qflow
                 Type = vaccineType
             };
 
-            var availableHours = qflowResponse.Availability
+            var availableSlotTimes = qflowResponse.Availability
                 .Select(x => x)
-                .OrderBy(x => x.Time.Hours)
-                .GroupBy(x => x.Time.Hours)
+                .OrderBy(x => x.Time)
+                .GroupBy(x => x.Time)
                 .Select(x => x.Key)
                 .ToList();
 
             var availabilityByHour = new List<AvailabilityHour>();
 
-            if (!availableHours.Any())
+            if (!availableSlotTimes.Any())
             {
                 response.AvailabilityByHour = availabilityByHour;
                 return response;
             }
 
-            foreach (var hour in availableHours)
+            var currentHour = currentDate.Hour;
+            var currentMinute = currentDate.Minute;
+
+            foreach (var slotTime in availableSlotTimes)
             {
+                if (slotTime.Hours < currentHour)
```

Comment:
Could this not be as simple as if slotTime < currentTime

Reply:
Ah it can. Have updated it